flow/flat_buffers.cpp

```diff
@@ -452,7 +452,7 @@ TEST_CASE("/flow/FlatBuffers/VectorRef") {
 			serializedVector = StringRef(readerArena, writer.toStringRef());
 		}
 		ArenaObjectReader reader(readerArena, serializedVector, Unversioned());
-		reader.deserialize(FileIdentifierFor<decltype(outVec)>::value, vecArena, outVec);
+		reader.deserialize(FileIdentifierFor<decltype(outVec)>::value, outVec, vecArena);
```

Comment:
Does it matter that the serialization order is arena then vector, while the deserialization order is vector then arena?

Reply:
It doesn't. Arenas are actually ignored altogether in terms of the wire protocol, so basically it's always a good idea to put the arena at the end in case any previous members have memory owned by the arena (which is probably why you're serializing the arena in the first place)

I think I'll add a comment to that effect so we're being more intentional about testing this behavior.